internal/compiler/fileloadertask.go

```diff
@@ -83,13 +85,20 @@ func (w *fileLoaderWorker[K]) start(loader *fileLoader, tasks []K, depth int, is
 			}
 
 			if startSubtasks {
-				w.start(loader, task.getSubTasks(), loadedTask.lowestDepth, task.isFromExternalLibrary())
+				w.start(loader, task.getSubTasks(), loadedTask.lowestDepth, loadedTask.fromExternalLibrary)
 			}
 		})
 	}
 }
 
 func (w *fileLoaderWorker[K]) collect(loader *fileLoader, tasks []K, iterate func(K, []tspath.Path)) []tspath.Path {
+	// Mark all tasks we saw as external after the fact.
+	w.tasksByFileName.Range(func(key string, value *queuedTask[K]) bool {
```

Comment:
Not quite what we do in strada right ? 

if the task is root it isnt marked as from external library - so dont you need that check ?

Also doing this in collect has possibility that it wont be marked as external if it was already collected? - though not sure if thats what we want or not